internal/core/monitoring/metrics_test.go

```diff
@@ -0,0 +1,131 @@
+//+build integration
+
+package monitoring
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCounterCreation(t *testing.T) {
```

Comment:
Here we're testing that our metrics are being registered, right? It feels like we're testing the Prometheus library. Could we use some process it offers to us? Like gathering metrics and checking if ours is there, we would avoid implementing a metrics server here and could move this test to `unit` since it does integrate only with an external library.

Also, I just want to keep in mind that these tests rely on some global/default states, so I'm not sure if those tests guarantee anything. For example, there is nothing preventing anyone from registering their metric with the same name as here, which would cause this test to fail because of external interference. (not sure how we could improve this tho, we could keep it this way and see if this is something we'll need to tackle later).

Reply:
I like the idea to transform it as unit. Still learning how to test it without calling it's API.